app/main.py

```diff
@@ -29,7 +29,7 @@ def __mul__(self, other: float | int | Distance) -> Distance:
     def __truediv__(self, other: float | int | Distance) -> Distance:
         return Distance(round((self.km / other), 2))
 
-    def __lt__(self, other: int | float) -> bool:
+    def __lt__(self, other: int | float | Distance) -> bool:
```

Comment:
if other is Distance instance you must add if isinstance(other, Distance): and self.km < other.km

add here and in all of bellow functions

Reply:
do it